### ISSUE TITLE
Create Download Page on Website

### ISSUE THREAD

**Issue by rmetzger** [stratosphere/stratosphere#220]
Hi,

I need someone to create a "Download" Page on our Website!

It should contain the various methods to get Stratosphere:
- Maven
- Binary Build
- From Sources (with description for profiles)
- The quickstarts
- Virtual Machine Demo
- Vagrant
- As a Debian Package  (not yet done)
- Using Puppet (not yet done)

There should be always two variants of each method. One for hadoop 1.2.1 and one for yarn.

Please open a Pull Request.

**Comment by rmetzger** [comment]
Assigned to @markus-h  (Sorry for assigning issues to you without talking to you first. I'll be in the office tomorrow. Or ask here.)

The binaries are here: http://dopa.dima.tu-berlin.de/bin/
The VMs are accessible from: http://dev.stratosphere.eu/vm/
The Vagrantfile is also there.
The scripts that generate the VMs are located in https://github.com/stratosphere/stratosphere-vms
How the maven snapshot repo and the two required are included into the pom is described in stratosphere-quickstart.

Debian-Packages and Puppet are not yet done.

**Comment by rmetzger** [comment]
Okay, we got a Debian package (which should be listed on the Downloads page too):

```
# vim /etc/apt/sources.list.d/stratosphere.list
deb http://dev.stratosphere.eu/repo binary/
```

```
# apt-get update
# apt-cache search stratosphere
stratosphere-dist - Sonatype helps open source projects to set up Maven repositories on https://oss.sonatype.org/
```

We should do some polishing on the package details:

```
Package: stratosphere-dist
Version: 0.2
Architecture: all
Maintainer: christian_richter@gmx.de
Installed-Size: 47558
Depends: java6-runtime
Filename: binary/stratosphere-dist_0.4-SNAPSHOT.deb
Size: 43797996
MD5sum: c9aaf70f1163f50503075124c9defb76
SHA1: e7e57a6f4974cd9aafc0f40ad11603ec71ec2423
SHA256: a895d1fcec80edea1b6d973c21930ff679123f401e933d8c5f1eb21b2ac5145a
Section: misc
Priority: low
Description: Sonatype helps open source projects to set up Maven repositories on https://oss.sonatype.org/
```

I think we should also test the debian package to see if it does everything as we want it.

**Comment by markus-h** [comment]
I have a first preview of the download page here: http://markus-h.github.io/stratosphere/download/

**Comment by asteriosk** [comment]
Hi Markus, most of the work is already there. 

Here are my comments: 
- I would say that we should keep this a real Download page. Meaning: you can remove the quickstart.
- I would avoid using a table because it makes the page complicated and cluttered. Try to keep things as they already are in the rest of the pages meaning: use headings.

If using headings, here is how I would structure the page:

= Download Page
Binary:
Stratosphere is compatible with Hadoop 1.2.x and also Hadoop V2. Please choose the right version that matches your environment etc...
 -- Hadoop 1.2
   -- Maven dependency 
 -- Hadoop 2
   -- Maven dependency 

For more information on how to use Stratosphere please refer to our "Quick Start" (/stratosphere/quickstart/)

Virtual Machines:
This is the quickest way to get up and running. Please go ahead and download one of the virtual machines ....etc
 -- VirtualBox
 -- Vagrant

Source:
 -- git clone ....
 -- Building with Hadoop 1.2 support: command inline
 -- Building with Hadoop 2 support:  command inline

What do you think?

**Comment by rmetzger** [comment]
I added some texts: http://robertmetzger.de/stratosphere/download/

**Comment by rmetzger** [comment]
So I have to admit, the table was my idea. You are right, it does not look very nice.

The reason why I initially suggested having a download page was that we need a central page that contains all the different ways to get Stratosphere. In the past, there had been many issues on github, some other places in the web etc. to get Stratosphere. This page should be a central point for everybody.

I don't know. How about searching the web for some download pages with similar requirements and see how they solved it.
For example: http://www.cloudera.com/content/support/en/downloads.html (I know we are an open source project, not a company)
http://www.eclipse.org/downloads/ (I like the 32, 64 bit solution there)
http://www.postgresql.org/download/ (this is the variant asterios suggested)
http://www.roda-community.org/download/ (nice icons, but too much text)

What do you think about my idea to step a bit back first.

**Comment by uce** [comment]
Hidiho,

I actually find our solution better (but maybe that's due to the medication I'm currently taking :dragon:). The other solutions are kind of overwhelming on a first visit.

(I was sick the last days, maybe I will propose something tonight to get back into business :smile:)

**Comment by uce** [comment]
This is my take on this. It doesn't resemble @rmetzger's proposals, but it could still be OK.

Preview: [http://uce.github.io](http://uce.github.io)

The sidebar on the left has a problem with the last sections, because they don't contain enough content (at most resolutions at least) for scrolling to "make sense".

The links are not functional. If we want this, I'll incorporate the changes and do a PR.

**Comment by aljoscha** [comment]
I prefer @uce's to the table based layout.

**Comment by asteriosk** [comment]
+1 for @uce's version! Great!

**Comment by uce** [comment]
OK, some further changes:
- Renamed Download to Downloads (with s) since we offer not a single but multiple files to download
- Added a text proposal from @StephanEwen to the startpage

Please check it out.

@rmetzger Please provide specific feedback on what needs to be polished (including real links for downloads etc.), so we can do the PR.

**Comment by rmetzger** [comment]
+1 @uce 

The Maven Deps are wrong.

``` xml
 <repositories>
   <repository>
     <id>snapshots-repo</id>
     <url>https://oss.sonatype.org/content/repositories/snapshots</url>
     <releases><enabled>false</enabled></releases>
     <snapshots><enabled>true</enabled></snapshots>
   </repository>
  </repositories>


<dependencies>
    <dependency>
      <groupId>eu.stratosphere</groupId>
      <artifactId>pact-common</artifactId>
      <version>0.4-SNAPSHOT</version>
    </dependency>
    <dependency>
      <groupId>eu.stratosphere</groupId>
      <artifactId>pact-clients</artifactId>
      <version>0.4-SNAPSHOT</version>
    </dependency>
  </dependencies>
```

Installation guide link is broken

The buttons seem to be quite large

Someone needs to test the debian package and installation process. This is untested and not automated (sebastian ruf is going to do that)

**Comment by StephanEwen** [comment]
Looks very good. When you corrected the maven dependencies, please push to the main branch.

**Comment by fhueske** [comment]
Like it!

**Comment by rmetzger** [comment]
Concerning the text on the start page "Stratosphere is a next-generation Big Data Analytics Platform. It combines the strenghts of MapReduce/Hadoop with powerful programming abstractions in Java and Scala and a high performance runtime. Stratosphere has native support for iterations, incremental iterations, and programs consisting of large DAGs of operations.".

I totally agree with having such a text, but: We should not mention iterations here. Most people hardly understand MapReduce, they only know that this is the big data thing, and they see that we are experts. Iterations are listed in the features below.

The features "normal people" understand (In my opinion):
- Better than Map Reduce, because we have more operators
- More than just Map and Reduce, complex data flows (people don't know what DAGs are)
- Very powerful programming interfaces: Java interface is like Cascading, Scala interface very fast to write code ("productive")

**Comment by uce** [comment]
OK, I agree. Can you provide an alternative text with your considerations? We should also have the text in sync with the README. Since the release is around the cornor, we should have this sooner than later.

**Comment by rmetzger** [comment]
Well, actually, I think we should remove everything from the Readme that is stated somewhere else! (check carefully!, I don't want to loose information)
It is hard to maintain documentation in so many places. (I noticed that when we changed the version)

(We should not say "is a next-generation" .. we are "THE next generation" ;))

"Stratosphere is the next-generation open source Big Data Analytics Platform. Extending MapReduce with new operators and supporting jobs with arbitrary complex operator graphs (more than just Map and Reduce). Programmable with a powerful Java interface and a very productive Scala API.
Stratosphere is compatible to existing Hadoop infrastructures but provides much better performance."

Just a suggestion.
